src/models/message_status.ts

```diff
@@ -18,22 +18,29 @@ import {
   RetrievedVersionedModel
 } from "../utils/cosmosdb_model_versioned";
 import { wrapWithKind } from "../utils/types";
+import { FiscalCode } from "../../generated/definitions/FiscalCode";
 
 export const MESSAGE_STATUS_COLLECTION_NAME = "message-status";
 export const MESSAGE_STATUS_MODEL_ID_FIELD = "messageId" as const;
 export const MESSAGE_STATUS_MODEL_PK_FIELD = "messageId" as const;
 
 // We cannot intersect with MessageStatus
 // as it is a *strict* interface
-export const MessageStatus = t.interface({
-  messageId: NonEmptyString,
-  status: MessageStatusValue,
-  updatedAt: Timestamp,
-  // eslint-disable-next-line sort-keys
-  isRead: withDefault(t.boolean, false),
-  // eslint-disable-next-line sort-keys
-  isArchived: withDefault(t.boolean, false)
-});
+export const MessageStatus = t.intersection([
+  t.interface({
+    messageId: NonEmptyString,
+    status: MessageStatusValue,
+    updatedAt: Timestamp,
+    // eslint-disable-next-line sort-keys
+    isRead: withDefault(t.boolean, false),
+    // eslint-disable-next-line sort-keys
+    isArchived: withDefault(t.boolean, false)
+  }),
+  // fiscalCode is optional due to retro-compatibility
+  t.partial({
```

Comment:
Not sure fiscalCode should be optional. 
Can we create a new MessageStatus model having fiscalCode as required and switching to this new model step by step (first step: use it for creation, second step: after batch operation ended, use it for reading too)? Not sure it's feasible @AleDore @michaeldisaro

Reply:
I think we could promote the fiscalCode to mandatory after the data cleaning.